src/index.ts

```diff
@@ -1,18 +1,21 @@
 import { Lottery } from "./lottery-application";
-import { writeFile } from 'fs';
-
-console.log("* Initializing Lottery Application *");
-
-var log = '***** Lottery application log *****\n\n';
+import * as fs from "fs";
+import {writeFile} from "fs";
 
 const upperBoundary = 60;
 const numbersToDraw = 6;
+const timestamp = new Date().toISOString();
```

Comment:
Ok, this needs a bit of discussion. In your opinion, should we initialise the timestamp in the beginning of the execution of the program or when we write the results?
From my experience, usually, since this program is **producing** data, we should put the timestamp at the end of the process.
Let me know what you think.

Reply:
All vars shall be declared at the beginning of the file/class. That is the standard in JS even though improvements have been made since ES6.
However, this does not exclude initialisation later in the script. So I agree, this can be moved where the execution is expected to start counting time. I will review this.
Thanks!

Reply:
I have tested this.
Here is the test I have done to confirm:
I have checked the value of the timestamp before the execution of the script (logging out to the console):
The value has been as follows:
**2022-10-14T16:01:08.485Z**
I have also logged the value of the text file created based on this timestamp:
**./results/lottery-results_2022-10-14T16:01:08.485Z.txt**

To verify if the value of the timestamp after the execution of the script differs from the initial value, I have logged the value of the same constant after writing to the file:
writeFile(fileName, log, () => console.log('Check result file ' + fileName));
console.log(timestamp, '4567');
console.log(fileName, '1111')

The value has been the same as initialised before the execution of the script:
**2022-10-14T16:01:08.485Z**
**./results/lottery-results_2022-10-14T16:01:08.485Z.txt** 

No need to do the code change here. The implementation suggested by me seems to be correct and in line with the approach in const/let declarations. Reasons for declaring all const/let/vars at the top of the class/file is due to previous unexpected issues with hoisting, which still exist if var is used instead of let/const. Initialisation of the timestamp at the same time do not affect its value during and after the execution of the script.

Please let me know about your thoughts.

Reply:
You are right.
I have tested with two different timestamps. Once before the execution of the script and one after the execution.
There is a difference:
**2022-10-14T16:17:29.515Z**
* Initializing Lottery Application *
** Running **
**2022-10-14T16:17:29.517Z**

I will correct the code accordingly and push the changes with the next commit.

Thanks for that! :)

Reply:
Issue corrected. Another commit pushed. Thanks!